ethexe/contracts/src/Middleware.sol

```diff
@@ -416,51 +362,127 @@ contract Middleware {
-        require(cfg.maxResolverSetEpochsDelay >= 3, "Resolver set epochs delay must be at least 3");
+        require(params.maxResolverSetEpochsDelay >= 3, "Resolver set epochs delay must be at least 3");
+    }
+
+    // TODO: check vault has enough stake
+    function _validateVault(address _vault) private {
+        if (!IRegistry(VAULT_REGISTRY).isEntity(_vault)) {
+            revert NonFactoryVault();
+        }
+
+        if (IMigratableEntity(_vault).version() != ALLOWED_VAULT_IMPL_VERSION) {
+            revert IncompatibleVaultVersion();
+        }
+
+        if (IVault(_vault).collateral() != COLLATERAL) {
+            revert UnknownCollateral();
+        }
+
+        /* Checking time */
+        uint48 vaultEpochDuration = IVault(_vault).epochDuration();
+        if (vaultEpochDuration < MIN_VAULT_EPOCH_DURATION) {
+            revert VaultWrongEpochDuration();
+        }
+
+        /* Validate delegator */
+        if (!IVault(_vault).isDelegatorInitialized()) {
+            revert DelegatorNotInitialized();
+        }
+
+        IBaseDelegator delegator = IBaseDelegator(IVault(_vault).delegator());
+        if (delegator.maxNetworkLimit(SUBNETWORK) != type(uint256).max) {
+            delegator.setMaxNetworkLimit(NETWORK_IDENTIFIER, type(uint256).max);
+        }
+        _delegatorHookCheck(IBaseDelegator(delegator).hook());
+
+        /* Validate Slasher */
+        if (!IVault(_vault).isSlasherInitialized()) {
+            revert SlasherNotInitialized();
+        }
+
+        address slasher = IVault(_vault).slasher();
+        if (IEntity(slasher).TYPE() != VETO_SLASHER_IMPL_TYPE) {
+            revert IncompatibleSlasherType();
+        }
+
+        if (IVetoSlasher(slasher).isBurnerHook()) {
+            revert BurnerHookNotSupported();
+        }
+
+        uint48 vetoDuration = IVetoSlasher(slasher).vetoDuration();
+        if (vetoDuration < MIN_VETO_DURATION) {
+            revert VetoDurationTooShort();
+        }
+
+        if (vetoDuration + MIN_SLASH_EXECUTION_DELAY > vaultEpochDuration) {
+            revert VetoDurationTooLong();
+        }
+
+        if (IVetoSlasher(slasher).resolverSetEpochsDelay() > MAX_RESOLVER_SET_EPOCHS_DELAY) {
+            revert ResolverSetDelayTooLong();
+        }
+
+        address resolver = IVetoSlasher(slasher).resolver(SUBNETWORK, new bytes(0));
+        if (resolver == address(0)) {
+            IVetoSlasher(slasher).setResolver(NETWORK_IDENTIFIER, VETO_RESOLVER, new bytes(0));
+        } else if (resolver != VETO_RESOLVER) {
+            // TODO: consider how to support this case
+            revert ResolverMismatch();
+        }
+
+        // TODO: consider allow transfer burned funds to ROUTER address
+        if (IVault(_vault).burner() == address(0)) {
+            revert UnsupportedBurner();
+        }
+    }
+
+    function _validateStakerRewards(address _vault, address _rewards) private view {
+        if (!IRegistry(STAKER_REWARDS_FACTORY).isEntity(_rewards)) {
```

Comment:
I think we should also add:
1) version == 1 check
2) reward contract "network middleware" check